lib/resourcemerge/batch.go

```diff
@@ -8,21 +8,26 @@ import (
 // EnsureJob ensures that the existing matches the required.
 // modified is set to true when existing had to be updated with required.
 func EnsureJob(modified *bool, existing *batchv1.Job, required batchv1.Job) {
-	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
 
-	if existing.Spec.Selector == nil {
-		*modified = true
-		existing.Spec.Selector = required.Spec.Selector
-	}
-	if !equality.Semantic.DeepEqual(existing.Spec.Selector, required.Spec.Selector) {
-		*modified = true
-		existing.Spec.Selector = required.Spec.Selector
+	// To set Selector you must set ManualSelector to True. If 'required'
+	// sets Selector to value not equal to existing, let current Job
+	// finish as Selector is immutable.
+	if required.Spec.ManualSelector != nil {
+
+		// If ManualSelector is not True we don't care about Selector
+		if *required.Spec.ManualSelector {
+			if !equality.Semantic.DeepEqual(existing.Spec.Selector, required.Spec.Selector) {
+				return
```

Comment:
I don't think we want this early `return`, because it's going to skip `Parallelism` and all the subsequent bits getting set later in `EnsureJob`.  Maybe you want the previous:

```go
*modified = true
existing.Spec.Selector = required.Spec.Selector
```

here?

Reply:
We can't change selector or we'll get the immutable error. This basically says, let the current job finish and then take care of all changes in next cycle UNLESS the selector just happens to be the same. Then we go ahead and break out and pick up all other changes.

Reply:
> We can't change selector or we'll get the immutable error.

Wouldn't that apply to all changes that set `modified` `true`?  Or is `selector` especially immutable?  For example, if you change the pod template, I'd have naively expected that to also need a delete/create Job.